engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/EqualsNode.java

```diff
@@ -32,9 +33,9 @@
       Does not throw dataflow errors or panics.
 
       Note that this is different than `Meta.is_same_object`, which checks whether two
-      references point to the same object on the heap.
-      """
-)
+      references point to the same object on the heap. However `Any.==` implies
+      `Meta.is_same_object` for all object with the exception of `Number.nan`.
+      """)
```

Comment:
This does not seem true, or am I misunderstanding?

If I have two separate instances of an object, they may be `==` but will not be `is_same_object`, right?

Shouldn't this condition be contraposed?

If I have `is_same_object x y` then I'm guranteed that `x == y`, as long as they are not `NaN`.

Reply:
You are right. I wrote the implication in the wrong direction. Fixed in d65e5e9f38a7b8b77e9673447af301744ff2d0a7.